lib/model/folder_sendrecv.go

```diff
@@ -455,25 +454,32 @@ nextFile:
 		// Check our list of files to be removed for a match, in which case
 		// we can just do a rename instead.
 		key := string(fi.Blocks[0].Hash)
-		for i, candidate := range buckets[key] {
+		var list []protocol.FileInfo
+		v := &protocol.Index{}
+		if ok := buckets.Get(key, v); ok {
+			list = v.Files
+		}
+		for i, candidate := range list {
 			if protocol.BlocksEqual(candidate.Blocks, fi.Blocks) {
 				// Remove the candidate from the bucket
-				lidx := len(buckets[key]) - 1
-				buckets[key][i] = buckets[key][lidx]
-				buckets[key] = buckets[key][:lidx]
+				lidx := len(list) - 1
+				list[i] = list[lidx]
+				list = list[:lidx]
+				buckets.Set(key, &protocol.Index{Files: list})
 
 				// candidate is our current state of the file, where as the
 				// desired state with the delete bit set is in the deletion
 				// map.
-				desired := fileDeletions[candidate.Name]
-				if err := f.renameFile(candidate, desired, fi, dbUpdateChan, scanChan); err != nil {
+				v := &protocol.FileInfo{}
```

Comment:
Generally in these kinds of operations, instead of 

```
v := &someType
it.Get(v)
use(*v)
```

I'd prefer doing the pointerizing explicitly instead of dereferencing, that is

```
var v someType
it.Get(&v)
use(v)
```

You also change a bunch of stuff from things like `chan protocol.FileInfo` to `chan *protocol.FileInfo` which may be justified but I'm not seeing the justification. One reason to *not* pass pointers is that once I got the value from the channel I know it's mine - there is no chance the sender has a reference and will inadvertently marshal into again behind my back or something like that.